docs/data/date-pickers/custom-field/JoyV6MultiInputRangeField.tsx

```diff
@@ -22,16 +22,13 @@ import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
```

Comment:
- Use `DateRangePickerFieldProps` (with a small adaptation to work with multi input range field, I can create a `DateRangePickerMultiInputFieldProps` if you want that in the lib, my assumption that since we migrate away from multi input, it's not needed).
- Remove `defaultValue` and `onError` props which are never passed by the picker to the field so a field built to be used inside a picker doesn't have to support them
- Add support for `unstableStartFieldRef` / `unstableEndFieldRef`